### PR TITLE
fix: mailjet source operating on array instead object

### PR DESCRIPTION
**Pull request by anantjain45823**
## What are the changes introduced in this PR?
All events received by rudderstack are coming in form of array of objects where as we are considering it as object only due which we are just sending empty payloads .
Nothing specifically is mention in mailjet docs about payload type
Mailjet Docs: https://dev.mailjet.com/email/guides/webhooks/#overview

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-1384

## Please explain the objectives of your changes below
To solve a basic type mismatch problem in mailjet source

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

## Summary by CodeRabbit

- **Refactor**
	- Enhanced event processing for improved data handling.

- **Tests**
	- Updated a test case to align with the new event processing logic.

### CODE REVIEW

File: src/v0/sources/mailjet/transform.js [hunk 1]
```diff
@@ -7,45 +7,49 @@ const Message = require('../message');
...
+      message.userId = md5(event.email);
+    }
+    messages.push(message);
+  });
+
+  return messages;
+};
+
+// This fucntion just converts the incoming payload to array of already not and sends it to processEvent
+const process = (event) => (Array.isArray(event) ? processEvent(event) : processEvent([event]));
```
Comment (koladilip) [inline src/v0/sources/mailjet/transform.js:53]: we could achieve the same thing with more clarity:

my converting original process function as processEvent which only process single event.
// using https://github.com/rudderlabs/rudder-workflow-engine/blob/bc179fb899892711e83e5e326dd69b2fd6f8e10c/src/bindings/common.ts#L30
const process = (events) => {
  const eventsArray = toArray(events);
  return eventsArray.map(process);
}

**Comment by yashasvibajpai** [comment]
This seems like a breaking change, and mailjet source looks to be broken since start then right @anantjain45823 ?

**Comment by anantjain45823** [comment]
Yes correct @yashasvibajpai you can refer to linear card for details

File: src/v0/sources/mailjet/transform.js (hunk 1 above)
Comment (anantjain45823) [reply src/v0/sources/mailjet/transform.js]: Done

**Review by koladilip — APPROVED** [review]

**Comment by sanpj2292** [comment]
@coderabbitai review

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2999/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2999/test-report.html

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**